### PR TITLE
Added the fix to make top 4 apis configurable 

### PR DESCRIPTION
**Pull request by manikey123**
#783

### CODE REVIEW

**Comment by manikey123** [comment]
@pmconrad @abitmore plz review the above pr for https://github.com/bitshares/bitshares-core/issues/783

**Comment by abitmore** [comment]
OP tagged #783. Actually #783 was not for making the limitations configurable, but for adding limitations to some APIs, which has been done by @oxarbitrage in #1272 for 2.0.190219 release (but we didn't close that issue, nor mention the API changes in release notes (which is not good since it has broken some client applications)). 

#782 was for changing the limitations to configurable, which is now closed, but we missed quite some APIs (mentioned in https://github.com/bitshares/bitshares-core/issues/783#issuecomment-486134761).

IMHO we can merge this PR. However, I tend to close #783 and reopen #782 and add the missing APIs to the list in #782. @pmconrad @ryanRfox what's your opinion? maybe create a new issue?

File: tests/common/database_fixture.cpp [hunk 1]
```diff
@@ -158,6 +158,27 @@ database_fixture::database_fixture()
     options.insert(std::make_pair("api-limit-get-key-references", boost::program_options::variable_value((uint64_t)200, false)));
     options.insert(std::make_pair("plugins", boost::program_options::variable_value(string("account_history"), false)));
    }
+   if(current_test_name =="api_limit_get_limit_orders")
+   {
+    options.insert(std::make_pair("api-limit-get-limit-orders", boost::program_options::variable_value((uint64_t)350, false)));
```
Comment (pmconrad) [inline tests/common/database_fixture.cpp]: Please wrap long lines at 113 chars or less.

File: tests/tests/database_api_tests.cpp [hunk 2]
```diff
@@ -971,4 +971,69 @@ BOOST_AUTO_TEST_CASE( api_limit_get_key_references ){
...
+   try{
+   graphene::app::database_api db_api( db, &( app.get_options() ));
+   //account_id_type() do 3 ops
+   create_bitasset("USD", account_id_type());
+   create_account("dan");
+   create_account("bob");
+   generate_block();
+   fc::usleep(fc::milliseconds(2000));
+   GRAPHENE_CHECK_THROW(db_api.get_call_orders(std::string(static_cast<object_id_type>(asset_id_type())), 370), fc::exception);
+   }catch (fc::exception& e) {
```
Comment (pmconrad) [inline tests/tests/database_api_tests.cpp]: Please add a positive test as well.

File: tests/tests/database_api_tests.cpp [hunk 3]
```diff
@@ -971,4 +971,69 @@ BOOST_AUTO_TEST_CASE( api_limit_get_key_references ){
...
+   try{
+   graphene::app::database_api db_api( db, &( app.get_options() ));
+   //account_id_type() do 3 ops
+   create_bitasset("USD", account_id_type());
+   create_account("dan");
+   create_account("bob");
+   generate_block();
+   fc::usleep(fc::milliseconds(2000));
+   GRAPHENE_CHECK_THROW(db_api.get_settle_orders(std::string(static_cast<object_id_type>(asset_id_type())), 370), fc::exception);
+   }catch (fc::exception& e) {
```
Comment (pmconrad) [inline tests/tests/database_api_tests.cpp]: same

File: tests/tests/database_api_tests.cpp [hunk 4]
```diff
@@ -971,4 +971,69 @@ BOOST_AUTO_TEST_CASE( api_limit_get_key_references ){
...
+   graphene::app::database_api db_api( db, &( app.get_options() ));
+   //account_id_type() do 3 ops
+   create_bitasset("USD", account_id_type());
+   create_account("dan");
+   create_account("bob");
+   asset_id_type bit_jmj_id = create_bitasset("JMJBIT").id;
+   generate_block();
+   fc::usleep(fc::milliseconds(2000));
+   GRAPHENE_CHECK_THROW(db_api.get_order_book(std::string(static_cast<object_id_type>(asset_id_type())), std::string(static_cast<object_id_type>(bit_jmj_id)),89), fc::exception);
+   }catch (fc::exception& e) {
```
Comment (pmconrad) [inline tests/tests/database_api_tests.cpp]: same

**Comment by pmconrad** [comment]
> what's your opinion? maybe create a new issue?

IMO create a new issue. Modifying the old ones will only add to the existing confusion. Thanks for sorting this out!

**Comment by oxarbitrage** [comment]
Pull currently having conflicts with merged htlc api limits. Will also conflict if https://github.com/bitshares/bitshares-core/pull/1749 gets merged before this one.

File: tests/tests/database_api_tests.cpp (hunk 3 above)
Comment (manikey123) [reply tests/tests/database_api_tests.cpp]: Done

File: tests/tests/database_api_tests.cpp (hunk 4 above)
Comment (manikey123) [reply tests/tests/database_api_tests.cpp]: done

File: tests/common/database_fixture.cpp (hunk 1 above)
Comment (manikey123) [reply tests/common/database_fixture.cpp]: done

**Comment by manikey123** [comment]
@pmconrad @abitmore plz review the updated PR

**Comment by pmconrad** [comment]
Changes look good.
Please rebase on latest develop to get rid of the conflicts.

**Comment by manikey123** [comment]
@pmconrad the changes are updated and Travis has built successfully. Plz review

**Comment by pmconrad** [comment]
There are more duplicates now, please double-check.

**Pull request merged**